pkg/ee/group-project-binding/handler/handler.go

```diff
@@ -136,3 +170,211 @@ func GetGroupProjectBinding(ctx context.Context, request interface{},
+}
+
+func (r createGroupProjectBindingReq) Validate() error {
+	if r.Body.Group == "" {
+		return utilerrors.NewBadRequest("`group` cannot be empty")
+	}
+
+	if err := validateRole(r.Body.Role); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func CreateGroupProjectBinding(ctx context.Context, request interface{},
+	userInfoGetter provider.UserInfoGetter,
+	projectProvider provider.ProjectProvider,
+	privilegedProjectProvider provider.PrivilegedProjectProvider,
+	bindingProvider provider.GroupProjectBindingProvider,
+) error {
+	req, ok := request.(createGroupProjectBindingReq)
+	if !ok {
+		return utilerrors.NewBadRequest("invalid request")
+	}
+
+	err := req.Validate()
+	if err != nil {
+		return utilerrors.NewBadRequest(err.Error())
+	}
+
+	kubermaticProject, err := common.GetProject(ctx, userInfoGetter, projectProvider, privilegedProjectProvider, req.ProjectID, nil)
+	if err != nil {
+		return common.KubernetesErrorToHTTPError(err)
+	}
+
+	userInfo, err := userInfoGetter(ctx, kubermaticProject.Name)
+	if err != nil {
+		return err
+	}
+
+	if err := bindingProvider.Create(ctx, userInfo, &kubermaticv1.GroupProjectBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("%s-%s", kubermaticProject.Name, rand.String(10)),
+		},
+		Spec: kubermaticv1.GroupProjectBindingSpec{
+			ProjectID: req.ProjectID,
+			Group:     req.Body.Group,
+			Role:      req.Body.Role,
+		},
+	}); err != nil {
+		return common.KubernetesErrorToHTTPError(err)
+	}
+	return nil
+}
+
+// swagger:parameters deleteGroupProjectBinding
+type deleteGroupProjectBindingReq struct {
+	groupProjectBindingReq
+}
+
+func DecodeDeleteGroupProjectBindingReq(c context.Context, r *http.Request) (interface{}, error) {
+	req, err := DecodeGroupProjectBindingReq(c, r)
+	if err != nil {
+		return nil, err
+	}
+	return deleteGroupProjectBindingReq{req.(groupProjectBindingReq)}, nil
+}
+
+func DeleteGroupProjectBinding(ctx context.Context, request interface{},
+	userInfoGetter provider.UserInfoGetter,
+	projectProvider provider.ProjectProvider,
+	privilegedProjectProvider provider.PrivilegedProjectProvider,
+	bindingProvider provider.GroupProjectBindingProvider,
+) error {
+	req, ok := request.(deleteGroupProjectBindingReq)
+	if !ok {
+		return utilerrors.NewBadRequest("invalid request")
+	}
+	kubermaticProject, err := common.GetProject(ctx, userInfoGetter, projectProvider, privilegedProjectProvider, req.ProjectID, nil)
+	if err != nil {
+		return common.KubernetesErrorToHTTPError(err)
+	}
+
+	userInfo, err := userInfoGetter(ctx, kubermaticProject.Name)
+	if err != nil {
+		return common.KubernetesErrorToHTTPError(err)
+	}
+
+	if err := bindingProvider.Delete(ctx, userInfo, req.BindingName); err != nil {
+		return common.KubernetesErrorToHTTPError(err)
+	}
+	return nil
+}
+
+// swagger:parameters patchGroupProjectBinding
+type patchGroupProjectBindingReq struct {
+	groupProjectBindingReq
+
+	// in: body
+	// required: true
+	Body struct {
+		Role string `json:"role"`
+	}
+}
```

Comment:
do we want to allow changing the Group?

Reply:
That was my initial thought. But I don't have strong opinion.

Reply:
I would allow, I dont see a reason why not